src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -464,6 +464,104 @@ public void It_doesnt_includes_requires_preview_features()
             Assert.False(contains);
         }
 
+        [Theory]
+        [InlineData(true, true, "net6.0", false)]
+        [InlineData(true, true, "net7.0", true)]
+        [InlineData(true, true, ToolsetInfo.CurrentTargetFramework, true)]
+        [InlineData(true, false, ToolsetInfo.CurrentTargetFramework, true)]
+        [InlineData(false, false, ToolsetInfo.CurrentTargetFramework, true)]
+        public void TestDisableRuntimeMarshalling(bool disableRuntimeMarshalling, bool generateDisableRuntimeMarshallingAttribute, string targetFramework, bool shouldHaveAttribute)
```

Comment:
In most cases the `GenerateDisableRuntimeMarshallingAttribute` property won't be explicitly set, so I think it would be clearer to use a nullable bool value for the parameter, and to leave it null when the test project shouldn't explicitly set it.